bindings/pyroot/pythonizations/python/ROOT/_pythonization/_rdf_pyz.py

```diff
@@ -0,0 +1,250 @@
+        n_cols = len(cols_list)
+        n_constants = len(extra_args)
+        
+        if n_cols>0: 
+            if n_params != n_cols + n_constants:# Check to see if all the parameters have been provided.
+                raise ValueError("Not Enough values provided in the column list and extra_args. The function required {} parameters only {} provided.".format(n_params, n_cols+n_constants))
+        
+        # Mapping the column list to the first input parameters of the function.
+        for idx, p in enumerate(self.params):
+            if idx < n_cols:
+                self.func_args[p] = cols_list[idx]
+        self.func_args = {**self.func_args, **extra_args} # Extra args supersedes col_list
+        return self
+
+    def find_function_signature(self):
+        """
+        Calculates the function signature. 
+        Updates the class with a new attribute args_info which contains the mapping of the parameter of the function to its type and value.
+        args_info:  dict{parameter_name(str): (type, value)}
+        """
+        func_args = self.func_args
+        for p in self.params:
+            if p in func_args: # the parameter value has been given in func_args
+                value_of_p = func_args[p]
+                type_of_p = self.find_type(value_of_p)
+                if type(value_of_p) == bool: # Bool(s) in python are represented as True/False but in C++ are true/false. The following if statements are to account for that
+                    if value_of_p: value_of_p = 'true'
+                    else: value_of_p = 'false'
+            else: # the parameter was not in func_args. Thus this parameter has to be mapped to a column of rdf
+                if p not in self.col_names:
+                    raise Exception(f"Unable to map function argument {p} to a column.\nUse correct name of column or pass a list of column names.")
+                value_of_p = p 
+                type_of_p = self.find_type(p)
+            self.args_info[p] = (type_of_p, value_of_p)
+        return self
+
+    def generate_function_call(self):
+        """
+        Generates a function call and signature. Gives a unique name to the function if it is a lambda function.
+        Updates the class with new attributes func_call and func_sign to hold them,
+        
+        Returns:
+            self: An instance of the class
+        """
+        func = self.func
+        if func.__name__ == '<lambda>' : 
+            func.__name__ = "_lambda_func_number" + str(FunctionJitter.lambda_function_counter)
+            FunctionJitter.lambda_function_counter += 1
+        func_call = func.__name__ + "("
+        func_sign = [str(self.args_info[p][0]) for p in self.params]
+        func_call = func_call + "".join(str(self.args_info[p][1]) + ", " for p in self.params)
+        if len(self.params) != 0:
+            func_call = func_call[:-2] + ")"
+        else:
+            func_call = func_call + ")"
+        self.func_call = func_call
```

Comment:
This can be just written as 
```suggestion
        self.func_call = f"{func.__name__}({','.join(arg_info[1] for arg_info in self.args_info.values())})"
```

If I have understood correctly the content of `self.args_info`

Reply:
Works just need a str conversiom to account for the non string arguments the are to be joined.
```python
        self.func_call = f"{func.__name__}({','.join(str(arg_info[1]) for arg_info in self.args_info.values())})"

Reply:
Can you describe with more details the contents of `self.args_info`? Why doesn't it contain a string?